lib/Sema/TypeCheckType.cpp

```diff
@@ -1131,21 +1131,26 @@ static Type diagnoseUnknownType(TypeResolution resolution,
       NominalTypeDecl *nominal = nullptr;
       if ((nominalDC = dc->getInnermostTypeContext()) &&
           (nominal = nominalDC->getSelfNominalTypeDecl())) {
-        // Attempt to refer to 'Self' within a non-protocol nominal
-        // type. Fix this by replacing 'Self' with the nominal type name.
-        assert(isa<ClassDecl>(nominal) && "Must be a class");
+        if (isa<ClassDecl>(nominal)) {
```

Comment:
We should do this for every type such that `!isa<ProtocolDecl>(nominal)`

Reply:
Which on this code path should be everything. We should relax the assert to just be that and drop the else here.

Reply:
You mean just replacing 
```diff
- if (isa<ClassDecl>(nominal)) {
+ if (!isa<ProtocolDecl>(nominal)) {
```

> Which on this code path should be everything. We should relax the assert to just be that and drop the else here.

Actually, in this test case, `nominal` is a `ProtocolDecl`. We just fail to find `Self` because the extension is not declared on the top level. One could argue that we should be able to find `Self` nonetheless but honestly, I don’t care too much for this bogus example in which we are already issuing a bunch of errors.